compiler/rustc_mir_build/src/thir/pattern/usefulness.rs

```diff
@@ -642,14 +642,7 @@ impl<'p, 'tcx> Usefulness<'p, 'tcx> {
                         .into_iter()
                         .flat_map(|witness| {
                             new_patterns.iter().map(move |pat| {
-                                Witness(
-                                    witness
-                                        .0
-                                        .iter()
-                                        .chain(once(pat))
-                                        .map(DeconstructedPat::clone_and_forget_reachability)
```

Comment:
Why is it okay not to forget reachability now? Is there a risk of having stale or incomplete reachability information?

Reply:
Nah this changes nothing, it's just that I hadn't implemented Clone to make sure no one cloned stuff by mistake. But it's inconvenient and there's not sensible situation where one would get confused. I have plans to clear that up more in the future